### PR TITLE
CHANGE (GithubActions) @W-15362391@ Modified Github Actions for flexibility.

### PR DESCRIPTION
**Pull request by jfeingold35**

### CODE REVIEW

File: github-actions/verify-pr-title/package.json [hunk 1]
```diff
@@ -17,13 +17,14 @@
 	},
```
Comment (stephen-carter-at-sf) [inline github-actions/verify-pr-title/package.json:17]: While you are in here, do you want to update our dependencies for this github action to the latest for @actions/core and @actions/github?

File: github-actions/verify-pr-title/src/index.ts [hunk 2]
```diff
@@ -19,30 +21,43 @@ function run(): void {
...
 		const base = pullRequest.base as {ref: string};
-		const targetBranch = base.ref;
-		if (targetBranch === "release" || targetBranch === "documentation") {
-			// Release branches have their own less stringent format.
-			if (verifyReleaseBranchPrTitle(title)) {
-				console.log(`PR title '${title}' accepted for release branch.`);
+		const baseBranch = base.ref;
+		const head = pullRequest.head as {ref: string};
+		const headBranch = head.ref;
+		if (headBranch === "main" && baseBranch === "dev") {
```
Comment (stephen-carter-at-sf) [inline github-actions/verify-pr-title/src/index.ts:27]: Do we care about dev-3?

File: github-actions/verify-pr-title/src/index.ts [hunk 3]
```diff
@@ -19,30 +21,43 @@ function run(): void {
...
+			// There's a title convention for merging `main` into `dev`.
+			if (verifyMain2DevPrTitle(title)) {
+				console.log(`PR title '${title}' accepted for dev branch.`);
 			} else {
 				core.setFailed(
-					`PR title '${title}' does not match the release PR template of "RELEASE: @W-XXXX@: Summary".`
+					`PR title '${title} does not match the template of "Main2Dev @W-XXXX@ Rebasing after vX.Y.Z"`
```
Comment (stephen-carter-at-sf) [inline github-actions/verify-pr-title/src/index.ts:33]: Did you intend to drop the 2nd single quote after `'${title}`?

File: github-actions/verify-pr-title/src/index.ts [hunk 4]
```diff
@@ -19,30 +21,43 @@ function run(): void {
...
+			// There's a title convention for merging PRs into `release`/`main`.
+			if (verifyReleasePrTitle(title)) {
+				console.log(`PR title '${title}' accepted for ${baseBranch} branch`);
+			} else {
+				core.setFailed(
+					`PR title '${title} does not match the template of "RELEASE @W-XXXX@ Summary"`
+				);
+				return;
+			}
+		} else if (baseBranch == "dev" || /^release-\d+\.\d+\.\d+$/.test(baseBranch)) {
```
Comment (stephen-carter-at-sf) [inline github-actions/verify-pr-title/src/index.ts:47]: Do we care about dev-3?

File: github-actions/verify-pr-title/src/common.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+ * All pull request titles will require this segment.
+ */
+export const WORK_ITEM_PORTION = "@W-\\d{4,9}@"
+
+/**
+ * This regex matches one or more whitespace, comma, period, n-dash,
+ * colon, or semicolon characters. All pull request titles allow these
+ * characters as separators between portions
+ */
+export const SEPARATOR = "[\\s-.,;:]+";
```
Comment (stephen-carter-at-sf) [inline github-actions/verify-pr-title/src/common.ts:14]: Do we want "\" or do we just want a space " ". I don't think we want tab or newline characters do we?

File: github-actions/verify-pr-title/package.json (hunk 1 above)
Comment (jfeingold35) [reply github-actions/verify-pr-title/package.json:17]: Yeah, no reason not to, I think.

File: github-actions/verify-pr-title/src/index.ts (hunk 2 above)
Comment (jfeingold35) [reply github-actions/verify-pr-title/src/index.ts:27]: No branch based off `dev` should ever be merged into `dev-3`, so there's no need to account for it here.
We could have a companion PR to this one that makes the equivalent changes in `dev-3` though, if we want.

File: github-actions/verify-pr-title/src/index.ts (hunk 3 above)
Comment (jfeingold35) [reply github-actions/verify-pr-title/src/index.ts:33]: Nope, that was a typo. Good catch.

File: github-actions/verify-pr-title/package.json (hunk 1 above)
Comment (jfeingold35) [reply github-actions/verify-pr-title/package.json:17]: Can easily update `@actions/core`. Updating `@actions/github` causes a compile failure that I'm looking into.

File: github-actions/verify-pr-title/src/index.ts (hunk 4 above)
Comment (jfeingold35) [reply github-actions/verify-pr-title/src/index.ts:47]: See above.

File: github-actions/verify-pr-title/src/common.ts (hunk 5 above)
Comment (jfeingold35) [reply github-actions/verify-pr-title/src/common.ts:14]: Yeah, I'm fine restricting it to space if we want.

**Review by stephen-carter-at-sf — APPROVED** [review]

**Pull request merged**